diff --git a/daemon/gdm-session-worker.c b/daemon/gdm-session-worker.c
index de3944149..c472ae180 100644
--- a/daemon/gdm-session-worker.c
+++ b/daemon/gdm-session-worker.c
@@ -2243,7 +2243,7 @@ gdm_session_worker_start_session (GdmSessionWorker  *worker,
 static gboolean
 set_up_for_new_vt (GdmSessionWorker *worker)
 {
-        int initial_vt_fd;
+        int initial_vt_fd, vt_0_fd;
         char vt_string[256], tty_string[256];
         int session_vt = 0;
 
@@ -2256,6 +2256,7 @@ set_up_for_new_vt (GdmSessionWorker *worker)
          * */
         g_snprintf (tty_string, sizeof (tty_string), "/dev/tty%d", GDM_INITIAL_VT);
         initial_vt_fd = open (tty_string, O_RDWR | O_NOCTTY);
+        vt_0_fd = open ("/dev/tty0", O_RDWR | O_NOCTTY);
 
         if (initial_vt_fd < 0) {
                 g_debug ("GdmSessionWorker: couldn't open console of initial fd: %m");
@@ -2275,6 +2276,8 @@ set_up_for_new_vt (GdmSessionWorker *worker)
                 }
         }
 
+        close (vt_0_fd);
+
         worker->session_vt = session_vt;
 
         g_assert (session_vt > 0);
@@ -2303,6 +2306,7 @@ set_up_for_new_vt (GdmSessionWorker *worker)
 
 fail:
         close (initial_vt_fd);
+        close (vt_0_fd);
         return FALSE;
 }
 